project_task_materials_stock/models/project.py

```diff
@@ -0,0 +1,180 @@
+# -*- coding: utf-8 -*-
+# Copyright 2015 Antiun Ingeniería S.L. - Sergio Teruel
+# Copyright 2015 Antiun Ingeniería S.L. - Carlos Dauden
+# Copyright 2016 Tecnativa - Vicent Cubells
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+from openerp import models, fields, api, exceptions, _
+
+
+class ProjectTaskType(models.Model):
+    _inherit = 'project.task.type'
+
+    consume_material = fields.Boolean(
+        string='Consume Material',
+        help="If you mark this check, when a task goes to this state,"
+             "it will consume the associated materials")
+
+
+class Task(models.Model):
+    _inherit = "project.task"
+
+    @api.multi
+    @api.depends('material_ids.stock_move_id')
+    def _compute_stock_move(self):
+        for task in self:
+            task.stock_move_ids = task.mapped('material_ids.stock_move_id')
+
+    @api.multi
+    @api.depends('material_ids.analytic_line_id')
+    def _compute_analytic_line(self):
+        for task in self:
+            task.analytic_line_ids = task.mapped(
+                'material_ids.analytic_line_id')
+
+    @api.multi
+    @api.depends('stock_move_ids.state')
+    def _compute_stock_state(self):
+        for task in self:
+            if not task.stock_move_ids:
+                task.stock_state = 'pending'
+            elif task.stock_move_ids.filtered(
+                    lambda r: r.state == 'confirmed'):
+                task.stock_state = 'confirmed'
+            elif task.stock_move_ids.filtered(lambda r: r.state == 'assigned'):
+                task.stock_state = 'assigned'
+            elif task.stock_move_ids.filtered(lambda r: r.state == 'done'):
+                task.stock_state = 'done'
```

Comment:
Probably this would be faster & shorter:

```
if not task.stock_move_ids:
    task.stock_state = 'pending'
else:
    states = task.mapped("stock_move_ids.state")
    for ("confirmed", "assigned", "done") as state:
        if state in states:
            task.stock_state = state
            break
```